src/core/edm/utils/useEntitySets.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * @flow
+ */
+
+import { Map, Set, isCollection } from 'immutable';
+import { Models } from 'lattice';
+import { LangUtils, ValidationUtils } from 'lattice-utils';
+import { useSelector } from 'react-redux';
+import type { UUID } from 'lattice';
+
+import {
+  EDM,
+  ENTITY_SETS,
+  ENTITY_SETS_INDEX_MAP,
+} from '../../redux/constants';
+
+const { isValidUUID } = ValidationUtils;
+const { isNonEmptyArray, isNonEmptyString } = LangUtils;
+const { EntitySet } = Models;
+
+type IdsOrNames =
+  | UUID[]
+  | string[]
+  | Set<UUID>
+  | Set<string>;
+
+// OPTIMIZE / MEMOIZE
```

Comment:
This would be a great candidate to try using https://github.com/reduxjs/reselect

Reply:
https://github.com/openlattice/lattice-orgs/pull/1084#discussion_r654156521